src/main/java/org/commonjava/indy/service/httprox/handler/ProxyRequestReader.java

```diff
@@ -72,6 +72,17 @@ public ProxyRequestReader(final ProxyResponseWriter writer, final ConduitStreamS
     public void handleEvent(final ConduitStreamSourceChannel sourceChannel) {
         boolean sendResponse = false;
         try {
+            if ( !sourceChannel.isOpen() )
+            {
+                return;
+            }
+
+            if ( !sourceChannel.isReadResumed() )
```

Comment:
When will this read-resume happen?

Reply:
It is at the beginning of the handler and a larger request ( from the codebase ) , but when running the test, the error still happens, so looks it does not help with this check.